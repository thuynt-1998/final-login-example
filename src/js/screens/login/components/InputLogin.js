import React from "react";
import { View, Text, TextInput as TextInputRender } from "react-native";
import { DefaultTheme, HelperText, TextInput } from "react-native-paper";
import { styles } from "../Login.style";
const theme = {
  ...DefaultTheme,
  colors: {
    ...DefaultTheme.colors,
    primary: "rgb(179,189,197)",
    text: "rgb(179,189,197)",
    placeholder: "rgb(179,189,197)",
  },
};
function InputLogin(props) {
  const { label, setValue, errors, title, left, secureTextEntry } = props;
  return (
    <View>
      <TextInput
        name={label}
        style={[styles.textInput, styles.textInputSuccess]}
        onChangeText={(text) => {
          setValue(label, text);
        }}
        theme={theme}
        left={<TextInput.Icon name={left} />}
        render={(props) => (
          <TextInputRender
            {...props}
            style={[styles.textInput, styles.textInputSucces]}
          />
        )}
        underlineColor="rgb(179,189,197)"
        placeholder={title}
        secureTextEntry={secureTextEntry}
      />
      <HelperText style={styles.errorInput}>
        {errors && errors.message}
      </HelperText>
    </View>
  );
}

export default InputLogin;
