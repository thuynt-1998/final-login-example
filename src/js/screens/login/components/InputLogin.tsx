import React, { memo } from "react";
import { View, Text, TextInput as TextInputRender } from "react-native";
import { DefaultTheme, HelperText, TextInput } from "react-native-paper";

import { styles } from "../Login.style";

const theme = {
  ...DefaultTheme,
  colors: {
    ...DefaultTheme.colors,
    primary: "rgb(179,189,197)",
    text: "rgb(179,189,197)",
    placeholder: "rgb(179,189,197)",
  },
};

interface PropsGlobal {
  label: string;
  onValue: (label: string, text: string) => any;
  errors: { message: string };
  title: string;
  left: () => React.ReactNode
  secureTextEntry: boolean;
}

const InputLogin = (props: PropsGlobal) => {
  const { label, onValue, errors, title, left, secureTextEntry } = props;

  return (
    <View style={{ width: "100%" }}>
      <TextInput
        accessibilityLabel="input"
        style={[styles.textInput, styles.textInputSuccess]}
        onChangeText={(text) => onValue(label, text)}
        theme={theme}
        left={<TextInput.Icon name={left} />}
        render={(props) => (
          <TextInputRender
            {...props}
            style={[styles.textInput]}
          />
        )}
        underlineColor="rgb(179,189,197)"
        placeholder={title}
        secureTextEntry={secureTextEntry}
      />
      {errors && <HelperText style={styles.errorInput} type="error" testID="nameError">
        {errors.message}
      </HelperText>}
    </View>
  );
}

export default memo(InputLogin);
