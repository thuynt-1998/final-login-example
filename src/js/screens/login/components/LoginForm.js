import React, { useCallback, useEffect, useState } from "react";
import {
  Text,
  View,
  Keyboard,
  TextInput as TextInputRender,
} from "react-native";
import { useForm } from "react-hook-form";
import { styles } from "../Login.style";
import { yupResolver } from "@hookform/resolvers/yup";
import { valid } from "../LoginForm.valid";
import { Button, TextInput } from "react-native-paper";
import InputLogin from "./InputLogin";
import {
  Ionicons,
  MaterialCommunityIcons,
  EvilIcons,
  Zocial,
  Fontisto,
} from "@expo/vector-icons";
import { themeLogin, themeFaceBook, themeGoogle } from "../theme";

function LoginForm(props) {
  const { register, handleSubmit, setValue, errors } = useForm({
    resolver: yupResolver(valid),
  });
  const { onLoginRequest } = props;
  const [isLogin, setIsLogin] = useState(false);
  useEffect(() => {
    register("username");
    register("password");
  }, [register]);
  const onClickLogin = useCallback((data) => {
    onLoginRequest(data.username, data.password);
    Keyboard.dismiss();
    setIsLogin(true);
  }, []);
  return (
    <View style={[styles.marginContainer, styles.flex1]}>
      <View style={[styles.flex1, styles.flexRow]}>
        <Text style={[styles.logo]}>digitm</Text>
      </View>
      <View style={{ flex: 2 }}>
        <InputLogin
          label="username"
          setValue={setValue}
          errors={errors.username}
          title="Email"
          left={() => (
            <Fontisto name="email" size={20} color="rgb(179,189,197)" />
          )}
          secureTextEntry={false}
        />
        <InputLogin
          label="password"
          setValue={setValue}
          errors={errors.password}
          title="Password"
          left={() => {
            return (
              <MaterialCommunityIcons
                name="shield-lock-outline"
                size={20}
                color="rgb(179,189,197)"
              />
            );
          }}
          secureTextEntry={true}
        />

        <Button
          mode="contained"
          style={[styles.button, styles.buttonRadius]}
          disabled={isLogin}
          onPress={handleSubmit(onClickLogin)}
          uppercase={false}
          labelStyle={[styles.fontSize16, styles.letterSpacing]}
          icon={() => {
            return (
              <Ionicons
                name="ios-lock"
                size={20}
                color={isLogin ? "rgba(0,0,0,0.3)" : "white"}
              />
            );
          }}
        >
          Secure Login
        </Button>

        <View style={[styles.flexRow, styles.marginVertical]}>
          <Text style={styles.text}>Don't have an account ? </Text>
          <Button
            theme={themeLogin}
            uppercase={false}
            style={styles.fontWeightOne}
            labelStyle={[styles.labelLink, styles.letterSpacing]}
          >
            Sign up Now!
          </Button>
        </View>

        <View style={styles.underline}>
          <View style={styles.underlineThought}></View>
          <Text style={[styles.underlineText, styles.text]}>OR</Text>
          <View style={styles.underlineThought}></View>
        </View>
        <Text style={[styles.text, styles.marginVertical]}>
          Sign in with Social Networks
        </Text>
        <View style={[styles.flexRow]}>
          <Button
            mode="contained"
            theme={themeFaceBook}
            style={[styles.flex1, styles.buttonRadius]}
            icon={() => (
              <EvilIcons name="sc-facebook" size={24} color="white" />
            )}
            uppercase={false}
            labelStyle={[styles.marginLeft5, styles.letterSpacing]}
          >
            Facebook
          </Button>
          <Button
            mode="contained"
            theme={themeGoogle}
            style={[styles.flex1, styles.marginLeft20, styles.buttonRadius]}
            icon={() => <Zocial name="google" size={20} color="white" />}
            uppercase={false}
            labelStyle={[styles.marginLeft5, styles.letterSpacing]}
          >
            Google
          </Button>
        </View>
      </View>
    </View>
  );
}

export default LoginForm;
