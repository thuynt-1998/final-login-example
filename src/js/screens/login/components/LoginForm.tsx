import React, { useCallback, useEffect, useState } from "react";
import { Text, View, Keyboard } from "react-native";
import { useForm } from "react-hook-form";
import { yupResolver } from "@hookform/resolvers/yup";
import { Button } from "react-native-paper";
import MaterialCommunityIcons from "react-native-vector-icons/MaterialCommunityIcons";
import EvilIcons from "react-native-vector-icons/EvilIcons";
import Zocial from "react-native-vector-icons/Zocial";
import Fontisto from "react-native-vector-icons/Fontisto";
import { useDispatch } from "react-redux";
import auth from '@react-native-firebase/auth';


import { styles } from "../Login.style";
import { valid } from "../LoginForm.valid";
import InputLogin from "./InputLogin";
import { themeLogin, themeFaceBook, themeGoogle } from "../theme";
import Creators from "../../../action";

const LoginForm = (props: { navigation: any }) => {
  const { register, handleSubmit, setValue, errors } = useForm({
    resolver: yupResolver(valid),
  });
  const [isLogin, setIsLogin] = useState(false);
  const dispatch = useDispatch();
  const onLoginRequest = useCallback((username: string, password: string) =>
    dispatch(Creators.loginRequest(username, password)), []
  );
  useEffect(() => {
    register("username");
    register("password");
  }, [register]);
  const onClickLogin = useCallback(({ username, password }: { username: string; password: string }) => {
    // onLoginRequest(data.username, data.password);
    auth()
      .signInWithEmailAndPassword(username, password)
      .then((res: any) => {
        console.log(res);
      })
      .catch((error: any) => {
        console.log(error);
      })
    Keyboard.dismiss();
    setIsLogin(true);
  }, []);
  const onValue = useCallback((name: string, value: string) => setValue(name, value), []);
  const onSignup = useCallback(() => props.navigation.push("signup"), [])
  return (
    <View style={[styles.marginContainer, styles.flex1]}>
      <View style={[styles.flex1, styles.flexRow]}>
        <Text style={[styles.logo]}>digitm</Text>
      </View>
      <View style={{ flex: 2 }}>
        <InputLogin
          label="username"
          onValue={onValue}
          errors={errors.username}
          title="Email"
          left={() => (
            <Fontisto name="email" size={20} color="rgb(179,189,197)" />
          )}
          secureTextEntry={false}
        />
        <InputLogin
          label="password"
          onValue={onValue}
          errors={errors.password}
          title="Password"
          left={() => {
            return (
              <MaterialCommunityIcons
                name="shield-lock-outline"
                size={20}
                color="rgb(179,189,197)"
              />
            );
          }}
          secureTextEntry={true}
        />
        <Button
          mode="contained"
          style={[styles.button, styles.buttonRadius]}
          disabled={isLogin}
          onPress={handleSubmit(onClickLogin)}
          uppercase={false}
          labelStyle={[styles.fontSize16, styles.letterSpacing]}
          icon={() => (
            <Fontisto
              name="locked"
              size={20}
              color={isLogin ? "rgba(0,0,0,0.3)" : "white"}
            />
          )
          }
        >
          Secure Login
        </Button>

        <View style={[styles.flexRow, styles.marginVertical]}>
          <Text style={styles.text}>Don't have an account ? </Text>
          <Button
            mode="text"
            theme={themeLogin}
            uppercase={false}
            labelStyle={[styles.labelLink, styles.letterSpacing, styles.fontWeightOne]}
            onPress={onSignup}
          >
            {"Sign up Now!"}
          </Button>
        </View>

        <View style={styles.underline}>
          <View style={styles.underlineThought}></View>
          <Text style={[styles.underlineText, styles.text]}>OR</Text>
          <View style={styles.underlineThought}></View>
        </View>
        <Text style={[styles.text, styles.marginVertical]}>
          Sign in with Social Networks
        </Text>
        <View style={[styles.flexRow]}>
          <Button
            mode="contained"
            theme={themeFaceBook}
            style={[styles.flex1, styles.buttonRadius]}
            icon={() => (
              <EvilIcons name="sc-facebook" size={24} color="white" />
            )}
            uppercase={false}
            labelStyle={[styles.marginLeft5, styles.letterSpacing]}
          >
            Facebook
          </Button>
          <Button
            mode="contained"
            theme={themeGoogle}
            style={[styles.flex1, styles.marginLeft20, styles.buttonRadius]}
            icon={() => <Zocial name="google" size={20} color="white" />}
            uppercase={false}
            labelStyle={[styles.marginLeft5, styles.letterSpacing]}
          >
            Google
          </Button>
        </View>
      </View>
    </View>
  );
}

export default LoginForm;
