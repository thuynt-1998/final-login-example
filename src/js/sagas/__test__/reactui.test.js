import React from "react";
import { fireEvent, render, act } from "@testing-library/react-native";
import { Provider } from "react-redux";
import { Fontisto } from "@expo/vector-icons";
import store from "../../store";
import LoginForm from "../../screens/login/components/LoginForm";
import InputLogin from "../../screens/login/components/InputLogin";

describe("test ui react native", () => {
  test("test InputLogin", () => {
    const onValue = jest.fn();
    const {
      getByLabelText,
      queryByTestId,
      getByTestId,
      getByPlaceholderText,
    } = render(
      <Provider store={store}>
        <InputLogin onValue={onValue} title="Email" />
      </Provider>
    );
    const { getByText } = render(
      <Provider store={store}>
        <LoginForm />
      </Provider>
    );
    const button = getByText("Secure Login");
    act(() => {
      fireEvent.press(button);
    });

    expect(queryByTestId("nameError")).toBeTruthy();
  });

  test("test InputLogin", () => {
    const onValue = jest.fn();
    const {
      getByLabelText,
      queryByTestId,
      getByTestId,
      getByPlaceholderText,
    } = render(
      <Provider store={store}>
        <InputLogin onValue={onValue} title="Email" />
      </Provider>
    );
    const input = getByPlaceholderText("Email");
    fireEvent.changeText(input, "thuy@gmail.com");
    const { getByText } = render(
      <Provider store={store}>
        <LoginForm />
      </Provider>
    );
    const button = getByText("Secure Login");
    act(() => {
      fireEvent.press(button);
    });

    expect(queryByTestId("nameError")).toBeTruthy();
  });
});
