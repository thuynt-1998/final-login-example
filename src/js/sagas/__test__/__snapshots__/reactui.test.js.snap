// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test ui react native test render component App 1`] = `undefined`;
